crates/edr_eth/src/block/reorg.rs

```diff
@@ -11,8 +11,9 @@ const DEFAULT_SAFE_BLOCK_TIME: Duration = Duration::from_secs(1);
 /// Test whether a block number is safe from a reorg for a specific chain based
 /// on the latest block number.
 pub fn is_safe_block_number(args: IsSafeBlockNumberArgs) -> bool {
-    let safe_block_number = largest_safe_block_number((&args).into());
-    args.block_number <= safe_block_number
+    args.latest_block_number
```

Comment:
@agostbiro could you please validate that the changes in this file are correct?

Changes were inspired by this code: https://github.com/NomicFoundation/hardhat/blob/edr/main/packages/hardhat-core/src/internal/hardhat-network/provider/utils/makeForkClient.ts#L172

Reply:
Would this return false for the latest block number if the latest block number is less than the safe depth? If yes, is that intended? I don't think that'd be aligned with `largest_safe_block_number` as that seems to return the latest block number if it's smaller than the safe depth.

Reply:
This function keeps returning whether it's safe. `largest_safe_block_number` now follows Hardhat's principle of return the latest block number, even if it's not safe.

Reply:
We discussed this on slack and decided to resolve by renaming `largest_safe_block_number`

Reply:
Addressed by https://github.com/NomicFoundation/hardhat/pull/4831/commits/985ffacb9216b38e216e0b04131f2e847fd0cbe1

I ended up doing a couple of things:
- I largely reverted the original change to keep a `largest_safe_block_number` function
- I changed the signature of the API to return an `Option<u64>`. This signals that an overflowing subtraction occurred. This was necessary to reflect that `is_safe_block_number` should be false for chains where the genesis block is unsafe too.
- I added a function `recommended_fork_block_number` in `edr_evm` that's used specifically to mimic the Hardhat design choice.